portal/src/test/scala/cromwell/pipeline/controller/ProjectControllerTest.scala

```diff
@@ -0,0 +1,128 @@
+        val userId = dummyUser.userId
+        val dummyProject = TestUserUtils.getDummyProject(userId)
+        val projectId = dummyProject.projectId
+        val accessToken = AccessTokenContent(userId.value)
+
+        when(projectService.getProjectById(dummyProject.projectId)).thenReturn(Future.successful(None))
+
+        Get(s"/projects/${projectId.value}") ~> projectController.route(accessToken) ~> check {
+          status shouldBe StatusCodes.NotFound
+        }
+      }
+    }
+
+    "DELETE project by id" should {
+      "return deactivated project entity" in {
+        val dummyUser: User = TestUserUtils.getDummyUser()
+        val userId = dummyUser.userId
+        val dummyProject = TestUserUtils.getDummyProject(userId)
+        val projectId = dummyProject.projectId
+        val accessToken = AccessTokenContent(userId.value)
+        val deactivatedProject = dummyProject.copy(active = false)
+
+        when(projectService.getProjectById(dummyProject.projectId)).thenReturn(Future.successful(Some(dummyProject)))
+        when(projectService.deactivateProjectById(dummyProject.projectId))
+          .thenReturn(Future.successful(Some(deactivatedProject)))
+
+        Delete(s"/projects/${projectId.value}") ~> projectController.route(accessToken) ~> check {
+          status shouldBe StatusCodes.OK
+          responseAs[Project] shouldBe deactivatedProject
+        }
+      }
+
+      "return '404 Not Found' if there is no project with such id" in {
+        val dummyUser: User = TestUserUtils.getDummyUser(active = false)
+        val userId = dummyUser.userId
+        val dummyProject = TestUserUtils.getDummyProject(userId)
+        val projectId = dummyProject.projectId
+        val accessToken = AccessTokenContent(userId.value)
+
+        when(projectService.getProjectById(dummyProject.projectId)).thenReturn(Future.successful(None))
+
+        Delete(s"/projects/${projectId.value}") ~> projectController.route(accessToken) ~> check {
+          status shouldBe StatusCodes.NotFound
+        }
+      }
+
+      "return '403 Forbidden' if user is not an owner of that project" in {
+        val dummyUser: User = TestUserUtils.getDummyUser()
+        val userId = dummyUser.userId
+        val dummyProject = TestUserUtils.getDummyProject(userId)
+        val projectId = dummyProject.projectId
+        val accessToken = AccessTokenContent("non-owner-userid")
+        val deactivatedProject = dummyProject.copy(active = false)
+
+        when(projectService.getProjectById(dummyProject.projectId)).thenReturn(Future.successful(Some(dummyProject)))
+        when(projectService.deactivateProjectById(dummyProject.projectId))
+          .thenReturn(Future.successful(Some(deactivatedProject)))
+
+        Delete(s"/projects/${projectId.value}") ~> projectController.route(accessToken) ~> check {
+          status shouldBe StatusCodes.Forbidden
+          responseAs[String] shouldBe "Project deactivation is allowed for owners only"
```

Comment:
Checking comparing string is not the best practice. 
There are two options: 
1. Go next with only status check -- that's ok
2. Extract string to the standalone variable (this variable should be used in tests and in business logic).
Second option is better, but the first one is also good.

Reply:
fixed